### PR TITLE
Devcon tweaks

### PR DESCRIPTION
**Pull request by aasseman**
Main things:
- Adding `NoisyCyclicSharedSubgraph`
- Fixing the plots, especially the Y scale "wobble" when query volume is noisy.
- Adding a bunch of experiment configs

### CODE REVIEW

File: simulation/multi_agent_simulation.py [hunk 1]
```diff
@@ -36,8 +36,8 @@ def main():
     ffmpeg_process = None
 
     # Environment x.
-    min_x = 1e-10
-    max_x = 5e-6
+    min_x = 1e-8
+    max_x = 8e-5
```
Comment (tkornuta-semiotic) [inline simulation/multi_agent_simulation.py:40]: we really need to move that to a config file or sth - not in this PR though

**Review by tkornuta-semiotic — DISMISSED** [review]
LGTM! :)

File: simulation/multi_agent_simulation.py (hunk 1 above)
Comment (aasseman) [reply simulation/multi_agent_simulation.py:40]: :100:

Comment (aasseman) [reply simulation/multi_agent_simulation.py:40]: #11

**Comment by aasseman** [comment]
Rebased onto latest master

**Pull request merged**